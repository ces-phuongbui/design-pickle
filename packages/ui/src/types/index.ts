interface Image {
  id: string;
  documentId: string;
  url: string;
  alternativeText: string | null;
  name: string;
}

export enum ButtonType {
  DEFAULT = "default",
  DESTRUCTIVE = "destructive",
  OUTLINE = "outline",
  SECONDARY = "secondary",
  GHOST = "ghost",
  LINK = "link",
}

export interface IButton {
  label: string;
  href?: string;
  type: ButtonType;
  color?: string;
}

export type ICarouselItem = {
  id: string | number;
  image: Image;
  name: string;
};

export type CardFeature = {
  id: number | string;
  title: string;
  cta: IButton;
  backgroundImage?: Image;
  description?: string;
};

export type CardStat = {
  value: string | number;
  label?: string;
  subLabel: string;
};

export type ComponentType =
  | "section.introduce-header"
  | "section.social-carousel"
  | "section.features"
  | "section.start";

interface Base<T extends ComponentType, D extends {} = {}> {
  __component: T;
  id: string;
  createdAt: string;
  updatedAt: string;
  data?: D;
}

export interface NavLink {
  href: string;
  text: string;
  isExternal: boolean;
  isPrimary: boolean;
}

export type Block = Introduce | SocialCarousel | Features | Stat;

export interface Introduce extends Base<"section.introduce-header"> {
  title: string;
  description: string;
  button: IButton;
}

export interface SocialCarousel extends Base<"section.social-carousel"> {
  bannerImage: Image;
  logos: ICarouselItem[];
}

export interface Features extends Base<"section.features"> {
  title: string;
  subTitle?: string;
  cards: CardFeature[];
}

export interface Stat extends Base<"section.start"> {
  title: string;
  subTitle: string;
  cards: CardStat[];
}
