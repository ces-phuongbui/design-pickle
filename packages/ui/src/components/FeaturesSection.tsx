import { Features } from "../types";
import { FeatureCard } from "./ui/feature-card";

const FeaturesSection = ({ title, subTitle, cards }: Features) => {
  return (
    <section className="bg-black py-16">
      <div className="container mx-auto px-4">
        <div className="mb-12 text-center">
          <h2 className="mb-4 text-4xl font-bold text-white">{title}</h2>
          {subTitle && <p className="text-sm text-gray-400">{subTitle}</p>}
        </div>

        <div className="grid gap-6 md:grid-cols-2 lg:grid-cols-3">
          {cards.map((card) => (
            <FeatureCard
              key={card.id}
              title={card.title}
              description={card.description ?? ""}
              backgroundImage={card.backgroundImage?.url ?? ""}
              ctaText={card.cta?.label}
              ctaHref={card.cta?.href}
            />
          ))}
        </div>
      </div>
    </section>
  );
};

export default FeaturesSection;
