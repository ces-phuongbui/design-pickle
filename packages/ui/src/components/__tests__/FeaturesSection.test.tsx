import { ButtonType, Features } from "@repo/ui/types/index";
import { render } from "@testing-library/react";
import FeaturesSection from "../FeaturesSection";

describe("FeaturesSection", () => {
  // Mock data for testing
  const mockFeatures: Features = {
    __component: "section.features",
    id: "1",
    createdAt: "2023-10-01T00:00:00.000Z",
    updatedAt: "2023-10-01T00:00:00.000Z",
    title: "Our Amazing Features",
    subTitle: "Explore what makes us unique",
    cards: [
      {
        id: 1,
        title: "Feature 1",
        description: "This is the first feature. It's awesome!",
        backgroundImage: {
          id: "img1",
          documentId: "doc1",
          url: "https://example.com/image1.jpg",
          alternativeText: "Feature 1 Image",
          name: "Feature 1",
        },
        cta: {
          label: "Learn More",
          href: "/feature1",
          type: ButtonType.LINK,
        },
      },
      {
        id: 2,
        title: "Feature 2",
        description: "This is the second feature. It's even better!",
        backgroundImage: {
          id: "img2",
          documentId: "doc2",
          url: "https://example.com/image2.jpg",
          alternativeText: "Feature 2 Image",
          name: "Feature 2",
        },
        cta: {
          label: "Get Started",
          href: "/feature2",
          type: ButtonType.DEFAULT,
        },
      },
      {
        id: 3,
        title: "Feature 3",
        description: "This is the third feature. It's the best!",
        cta: {
          label: "Discover",
          href: "/feature3",
          type: ButtonType.OUTLINE,
        },
      },
    ],
  };

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<FeaturesSection {...mockFeatures} />);

    // Match snapshot
    expect(container).toMatchSnapshot();
  });

  it("renders correctly without background images", () => {
    const propsWithoutBackground: Features = {
      __component: "section.features",
      id: "1",
      createdAt: "2023-10-01T00:00:00.000Z",
      updatedAt: "2023-10-01T00:00:00.000Z",
      title: "Our Amazing Features",
      subTitle: "Explore what makes us unique",
      cards: [
        {
          id: 1,
          title: "Feature 1",
          description: "This is feature 1",
          cta: {
            label: "Learn More",
            href: "/feature1",
            type: ButtonType.DEFAULT,
          },
        },
      ],
    };

    const { container, getByText } = render(
      <FeaturesSection {...propsWithoutBackground} />,
    );

    // Check if the feature card is rendered without a background image
    expect(getByText("Feature 1")).toBeInTheDocument();
    expect(getByText("This is feature 1")).toBeInTheDocument();

    // Match snapshot
    expect(container).toMatchSnapshot();
  });
});
