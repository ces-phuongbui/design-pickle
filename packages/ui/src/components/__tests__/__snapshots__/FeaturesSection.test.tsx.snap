// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesSection renders correctly and matches snapshot 1`] = `
<div>
  <section
    class="bg-black py-16"
  >
    <div
      class="container mx-auto px-4"
    >
      <div
        class="mb-12 text-center"
      >
        <h2
          class="mb-4 text-4xl font-bold text-white"
        >
          Our Amazing Features
        </h2>
        <p
          class="text-sm text-gray-400"
        >
          Explore what makes us unique
        </p>
      </div>
      <div
        class="grid gap-6 md:grid-cols-2 lg:grid-cols-3"
      >
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <img
              alt="Feature 1"
              class="h-full w-full object-cover"
              data-nimg="1"
              decoding="async"
              height="600"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=828&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 1
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the first feature. It's awesome!
              </p>
              <a
                href="/feature1"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Learn More
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <img
              alt="Feature 2"
              class="h-full w-full object-cover"
              data-nimg="1"
              decoding="async"
              height="600"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=828&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 2
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the second feature. It's even better!
              </p>
              <a
                href="/feature2"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Get Started
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <div
              class="absolute inset-0 bg-gradient-to-b from-transparent via-purple-600/50 to-teal-900/90"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 3
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the third feature. It's the best!
              </p>
              <a
                href="/feature3"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Discover
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`FeaturesSection renders correctly with all props 1`] = `
<div>
  <section
    class="bg-black py-16"
  >
    <div
      class="container mx-auto px-4"
    >
      <div
        class="mb-12 text-center"
      >
        <h2
          class="mb-4 text-4xl font-bold text-white"
        >
          Our Amazing Features
        </h2>
        <p
          class="text-sm text-gray-400"
        >
          Explore what makes us unique
        </p>
      </div>
      <div
        class="grid gap-6 md:grid-cols-2 lg:grid-cols-3"
      >
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <img
              alt="Feature 1"
              class="h-full w-full object-cover"
              data-nimg="1"
              decoding="async"
              height="600"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=828&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 1
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the first feature. It's awesome!
              </p>
              <a
                href="/feature1"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Learn More
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <img
              alt="Feature 2"
              class="h-full w-full object-cover"
              data-nimg="1"
              decoding="async"
              height="600"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=828&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 2
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the second feature. It's even better!
              </p>
              <a
                href="/feature2"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Get Started
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <div
              class="absolute inset-0 bg-gradient-to-b from-transparent via-purple-600/50 to-teal-900/90"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 3
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is the third feature. It's the best!
              </p>
              <a
                href="/feature3"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Discover
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`FeaturesSection renders correctly without background images 1`] = `
<div>
  <section
    class="bg-black py-16"
  >
    <div
      class="container mx-auto px-4"
    >
      <div
        class="mb-12 text-center"
      >
        <h2
          class="mb-4 text-4xl font-bold text-white"
        >
          Our Amazing Features
        </h2>
        <p
          class="text-sm text-gray-400"
        >
          Explore what makes us unique
        </p>
      </div>
      <div
        class="grid gap-6 md:grid-cols-2 lg:grid-cols-3"
      >
        <div
          class="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
          data-testid="feature-card"
        >
          <div
            class="absolute inset-0 z-0"
          >
            <div
              class="absolute inset-0 bg-gradient-to-b from-transparent via-purple-600/50 to-teal-900/90"
            />
          </div>
          <div
            class="relative z-10 flex h-full flex-col justify-between"
          >
            <h3
              class="text-3xl font-bold"
            >
              Feature 1
            </h3>
            <div
              class="space-y-4"
            >
              <p
                class="text-xl text-gray-100 mb-5"
              >
                This is feature 1
              </p>
              <a
                href="/feature1"
              >
                <button
                  class="inline-flex items-center justify-center rounded-md text-sm font-semibold transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background underline-offset-4 h-10 group text-white hover:no-underline bg-white bg-opacity-20 p-4"
                >
                  Learn More
                   
                  <span
                    class="ml-2 transition-transform group-hover:translate-x-1"
                  >
                    →
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
